### PR TITLE
Base implementation

### PR DESCRIPTION
**Pull request by vjik**

### CODE REVIEW

File: src/Domain/Account/AccountChartId.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpFinance\DoubleEntry\Domain\Account;
+
+final readonly class AccountChartId
```
Comment (samdark) [inline src/Domain/Account/AccountChartId.php:7]: What's the benefit of having a class for it?

File: src/Domain/Account/AccountRepositoryInterface.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+use PhpFinance\DoubleEntry\Domain\Account\Exception\AccountNotFoundException;
+
+interface AccountRepositoryInterface
+{
+    /**
+     * @throws AccountNotFoundException
+     */
+    public function get(AccountId $id): Account;
+
+    public function existsChildren(Account $account): bool;
```
Comment (samdark) [inline src/Domain/Account/AccountRepositoryInterface.php:16]:
```suggestion
    public function hasChildren(Account $account): bool;
```

Comment (vjik) [reply src/Domain/Account/AccountRepositoryInterface.php]: done

File: src/Domain/Account/AccountChartId.php (hunk 1 above)
Comment (vjik) [reply src/Domain/Account/AccountChartId.php:7]: Users of `php-finance/double-entry` will use this object as ID of account chart.

File: src/Domain/Transaction/Factory/TransactionFactory.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    public function __construct(
+        private TransactionIdFactory $transactionIdFactory,
+        private PostingFactory $postingFactory,
+    ) {
+    }
+
+    public function create(
+        DateTimeImmutable $date,
+        PostingData $postingData,
+        PostingData ...$postingsData,
```
Comment (samdark) [inline src/Domain/Transaction/Factory/TransactionFactory.php:22]: Looks interesting. What's intent here?

Comment (vjik) [reply src/Domain/Transaction/Factory/TransactionFactory.php:22]: This ensures that at least one posting will be passed.

File: src/Domain/Posting/Exception/DimensionNotFoundException.php [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+use LogicException;
+
+final class DimensionNotFoundException extends LogicException
+{
+    public function __construct(int|string $name)
+    {
+        parent::__construct(
+            sprintf(
+                'Entry do not contain dimension "%s".',
```
[the remark this answers is not in the document]
Comment (vjik) [reply src/Domain/Posting/Exception/DimensionNotFoundException.php]: done

File: src/Domain/Account/AccountRepositoryInterface.php [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+     * @throws AccountNotFoundException
+     */
+    public function get(AccountId $id): Account;
+
+    public function hasChildren(Account $account): bool;
+
+    public function save(Account $account): void;
+
+    /**
+     * Don't run directly, use {@see AccountManager::delete()} instead of.
```
[the remark this answers is not in the document]
Comment (vjik) [reply src/Domain/Account/AccountRepositoryInterface.php:21]: Account denied to deletions in so cases:

- account has children accounts;
- has entries with account.

`AccountManager::delete()` will be check this conditions.

**Pull request merged**